src/pyimpspec/data/formats/par.py

```diff
@@ -0,0 +1,91 @@
+# pyimpspec is licensed under the GPLv3 or later (https://www.gnu.org/licenses/gpl-3.0.html).
+# Copyright 2023 pyimpspec developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# The licenses of pyimpspec's dependencies and/or sources of portions of code are included in
+# the LICENSES folder.
+
+from os.path import exists
+from typing import (
+    IO,
+    List,
+)
+from pyimpspec.data.data_set import (
+    DataSet,
+    dataframe_to_data_sets,
+)
+
+
+def parse_par(path: str) -> List[DataSet]:
+    """
+    Parse a Parstat .par file containing an impedance spectrum.
+
+    Parameters
+    ----------
+    path: str
+        The path to the file to process.
+
+    Returns
+    -------
+    List[DataSet]
+    """
+    from pandas import DataFrame
+    assert isinstance(path, str) and exists(path), path
+    fp: IO
+    with open(path, "r", encoding="latin1") as fp:
+        lines: List[str] = list(
+            filter(lambda _: _ != "", map(str.lower, map(str.strip, fp.readlines())))
+        )
+
+    assert len(lines) > 0, f"Failed to find any impedance data in '{path}'"  # check for blank file
+    # Need to make sure we have the correct Action for the EIS experiment
+    found_eis = False
+    eis_action_no = None
+    while not found_eis:
+        line: str = lines.pop(0)
+        if line.startswith("<action"):
+            eis_action_no = str(int(line[7:][:-1]) - 1)  # Parstat's action segment numbering is off by one for some reason.
```

Comment:
I'm not familiar with VersaStudio or the .par file format, but it seems like there can be multiple measurements. Can these measurements be, e.g., OCP -> CV -> EIS? Or just repeats of the same measurement type?

Reply:
What about a scenario where a single file contains results from multiple EIS experiments? Only one `Segment` can be parsed at the moment.

It would be great if you could confirm that the current principle of the approach to determining whether or not a line should be parsed (i.e., line 70) would also work when there are multiple EIS experiments to parse. Your comment on line 59 states that the numbering is off by one. In the example data file that you included, the value of `X` in `<ActionX>` and `<SegmentX>` match (in this case `1`) for the EIS experiment but the values in the `Segment #` column inside the `<Segment1>` section do seem to be decremented by one. I say *seem*, because there is also an `<Action0>` section that is missing a corresponding `<Segment0>` section. Also, note that there is a `Segments=1` line in the `<Action1>` section while the corresponding line in the `<Action0>` section is `Segments=0`. So the values in the `Segment #` may be referring to another set of indices.

Reply:
I'm afraid the way VersaStudio works is it allows an arbitrary number of segments based on how the user sets it up. Unfortunately, they aren't always defined the same way in the `<ActionX>` sections. I have files from experiments I've run that combine open-circuit measurements over time, then EIS, then some polarization testing. The corresponding `<ActionX>` sections only seem to have the `Segments=X` statement for the EIS and for the initial `<Action0>` block. And to make it worse, some LPR experiments have 3 different `<Segment>` blocks that are just incrementally numbered based on the chronological order they're run. I can think about the best way to pull out multiple EIS data sets from a file for pyimpspec integration. It may be just as simple as counting the number of EIS `<Action>` blocks in the file and making sure the correct number of `<Segment>` sections are parsed.

Reply:
I found a copy of a relatively recent manual for VersaStudio and it states that the EIS data stored in the raw data files is uncalibrated and should not be used for data analysis. The data should first be opened in VersaStudio and exported to, e.g., ZView or CSV prior to analysis.

Reply:
Yes I agree it is strange. I don't particularly care for VersaStudio, though the potentiostat itself is fantastic. My general work flow is to export the data to ASCII format, then use pyimpspec for analysis. The exported data still (so far) have a .par extension. The raw data files are in a binary format that isn't easily opened, though they still have a .par extension. which is not great in my opinion.

We have had issues with the "calibration correction" that VersaStudio does in the past. For example, I've performed an anodic scan up to, say +1V vs OCP, but when I go to analyze the data the curve goes up to +15 V. So I have to go back, re-open the data in VersaStudio, and export again. It can be somewhat frustrating if you don't notice the issue in VersaStudio. To your point, if it applies new calibrations to old data, then that will of course be problematic as well.